src/main/java/roomescape/web/controller/ControllerExceptionHandler.java

```diff
@@ -0,0 +1,52 @@
+package roomescape.web.controller;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.dao.DataAccessException;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+import roomescape.web.exception.CustomException;
+
+@RestControllerAdvice
+public class ControllerExceptionHandler {
+    private final Logger log = LoggerFactory.getLogger(getClass());
+
+    @ExceptionHandler(CustomException.class)
+    public ResponseEntity<String> handleCustomException(final CustomException exception) {
+        log.error("[Custom Exception]", exception);
+        return new ResponseEntity<>(exception.getMessage(), exception.getStatus());
+    }
+
+    @ExceptionHandler(MethodArgumentNotValidException.class)
+    public ResponseEntity<String> handleMethodArgumentNotValidException(
+            final MethodArgumentNotValidException exception
+    ) {
+        log.error("[Method Argument Not Valid Exception]", exception);
+        String message = exception.getBindingResult()
+                .getAllErrors()
+                .get(0)
+                .getDefaultMessage();
+        return new ResponseEntity<>(message, HttpStatus.BAD_REQUEST);
+    }
+
+    @ExceptionHandler(IllegalArgumentException.class)
+    public ResponseEntity<String> handleIllegalArgumentException(final IllegalArgumentException exception) {
+        log.error("[Illegal Argument Exception]", exception);
+        return new ResponseEntity<>("잘못된 파라미터 요청입니다.", HttpStatus.BAD_REQUEST);
+    }
+
+    @ExceptionHandler(DataAccessException.class)
+    public ResponseEntity<String> handleDataAccessException(final DataAccessException exception) {
```

Comment:
```suggestion
    public ResponseEntity<ErrorResponse> handleDataAccessException(final DataAccessException exception) {
```

String을 그대로 내려주기보다. ErrorResponse 클래스를 만들어서 json형태로 값이 내려가도록 하면 더 좋을것 같습니다!
다른 부분도 그렇게 처리되면 좋을것 같네요!

Reply:
json 형태로 전달하도록 수정했습니다:)
springframework에서 제공하는 `ProblemDetail` 클래스가 있어 사용했습니다!

응답 예시:
```json
{
    "type": "about:blank",
    "title": "Authorization Exception",
    "status": 401,
    "detail": "토큰이 없거나 만료되었습니다.",
    "instance": "/admin/reservation"
}
```